core/extension/gdextension.cpp

```diff
@@ -537,7 +537,13 @@ Ref<Resource> GDExtensionResourceLoader::load(const String &p_path, const String
 
 	Ref<GDExtension> lib;
 	lib.instantiate();
+#ifdef ANDROID_ENABLED
+	// On Android, the libraries are placed in the lib folder of the apk,
+	// so we only use relative paths here.
+	String abs_path = library_path.get_file();
```

Comment:
This would be good to get confirmation from the android team for, but I guess this makes sense

Reply:
To add to this, this logic makes sense for gdextension packaged within Android apks. 

The logic will need to be updated for the Godot Android Editor since it should be able to load gdextension plugins outside of its own apk. I can take a look in making it work after this pr is merged.

Reply:
@dsnopek FYI, this means that gdextension has been broken on Android up to now.
This is something we may want to include in the Godot 4.2 release notes.